lisp/ess-inf.el

```diff
@@ -1293,43 +1293,44 @@ wrapping the code into:
         ;; Swap the process buffer with the output buffer before
         ;; sending the command
         (unwind-protect
-            (progn
-              (when use-sentinel
-                (process-put proc 'ess-output-sentinel sentinel))
-              (set-process-buffer proc out-buffer)
-              (set-process-filter proc 'inferior-ess-ordinary-filter)
-              (with-current-buffer out-buffer
-                (ess-setq-vars-local proc-alist)
-                (setq buffer-read-only nil)
-                (erase-buffer)
-                (set-marker (process-mark proc) (point-min))
-                (inferior-ess-mark-as-busy proc)
-                (process-send-string proc rich-cmd)
-                ;; Need time for ess-create-object-name-db on PC
-                (if no-prompt-check
-                    (sleep-for 0.02)   ; 0.1 is noticeable!
-                  (unless (ess-wait-for-process proc nil wait force-redisplay timeout)
-                    (error "Timeout during background ESS command `%s'"
-                           (ess--strip-final-newlines cmd)))
-                  ;; Remove prompt. If output is cat(..)ed without a
-                  ;; final newline, this deletes the last line of output.
+            (unwind-protect
+                (progn
```

Comment:
How about adding unwind-protect just on the sentinel+ess-interrupt block instead? Seems cleaner and easier to reason about to me.

Reply:
IIUC that was how I initially wrote it but something fell off using unwind-protect in the protect block. I tried to google if it was a normal pattern but couldnt find anything. Happy to change it back if you think it's better.

Reply:
I think it's cleaner as it can be nested into `(when (and use-sentinel early-exit) ...)` block. We don't really need `unwind-protect` there, simple `condition-case-unless-debug` would do, no?